pkg/pbq/store/interfaces.go

```diff
@@ -0,0 +1,19 @@
+package store
+
+import (
+	"github.com/numaproj/numaflow/pkg/isb"
+)
+
+// Store provides methods to read, write and delete data from a durable store.
+type Store interface {
+	// ReadFromStore returns all the messages from the persisted store
+	ReadFromStore(size int64) ([]*isb.Message, error)
```

Comment:
I assume the implementation of this function will have to remember the read offset right? Thinking if using a channel read would simplify the logic....